sycl/include/CL/sycl/handler.hpp

```diff
@@ -731,14 +731,19 @@ class __SYCL_EXPORT handler {
 
     // FIXME Remove the ESIMD check once rounding of execution range works well
     // with ESIMD compilation flow.
+    // Range rounding can be disabled by the user.
+    // Range rounding is not done on the host device.
     // Range rounding is supported only for newer SYCL standards.
-    // Range rounding can also be disabled by the user.
 #if !defined(__SYCL_EXPLICIT_SIMD__) &&                                        \
     !defined(SYCL_DISABLE_PARALLEL_FOR_RANGE_ROUNDING) &&                      \
-    SYCL_LANGUAGE_VERSION >= 202001
-    // The work group size preferred by this device.
-    // A reasonable choice for rounding up the range is 32.
-    constexpr size_t GoodLocalSizeX = 32;
+    !defined(DPCPP_HOST_DEVICE_OPENMP) &&                                      \
+    !defined(DPCPP_HOST_DEVICE_PERF_NATIVE) && SYCL_LANGUAGE_VERSION >= 202001
+    // Range should be a multiple of this for reasonable performance.
+    constexpr size_t MinFactorX = 16;
```

Comment:
Can we have these values also configurable through environment variables?
The current values can be used as default.